include/slang/binding/AssignmentExpressions.h

```diff
@@ -65,15 +65,19 @@ class AssignmentExpression : public Expression {
 struct CastExpressionSyntax;
 struct SignedCastExpressionSyntax;
 
+enum class ConvertCast : uint8_t { Propagate, Implicit, BitstreamCast, Explicit };
+
 /// Represents a type conversion expression (implicit or explicit).
 class ConversionExpression : public Expression {
 public:
-    bool isImplicit;
+    bool isImplicit() const { return cast_ <= ConvertCast::Implicit; }
+    void setCast(ConvertCast);
+    ConvertCast cast() const { return cast_; }
 
     ConversionExpression(const Type& type, bool isImplicit, Expression& operand,
```

Comment:
Can you make this take the ConversionKind directly during construction? No need for the boolean anymore.

Reply:
This is possible, but requires more messy change on existing code. The two ConversionExpression::fromSyntax functions at lines 522 and 470 of AssignmentExpressions.cpp both construct ConversionExpression first, then analyze compatibility,  return badExpr if not compatible. In other words, the ConversionKind data is available after construction. To make ConversionKind available during construction, the code logic in the two fromSyntax function needs more changes.
My original intention is to minimize modification of existing good code. If you still think it is fine to change the two fromSyntax functions significantly, I can make the changes.

Reply:
Yeah, I think that's fine. Having class properties be immutable after construction is pretty valuable, worth some refactoring I think.

Reply:
Yes, I initialize ConversionKind in in the constructor and it does not need a change later. The refactoring of fromSyntax functions turns out to be simple. I only need to modify a few expressions and the existing code logic and flow remain intact.